tools/runners/run-mongo.js

```diff
@@ -585,21 +585,24 @@ var launchMongo = function (options) {
   var initiateReplSetAndWaitForReady = function () {
     try {
       // Load mongo so we'll be able to talk to it.
-      const { Db, Server } = loadIsopackage('npm-mongo').NpmModuleMongodb;
+      const {
+        MongoClient,
+        Server
+      } = loadIsopackage('npm-mongo').NpmModuleMongodb;
 
       // Connect to the intended primary and start a replset.
-      var db = new Db(
-        'meteor',
+      const client = new MongoClient(
         new Server('127.0.0.1', options.port, {
           poolSize: 1,
           socketOptions: {
             connectTimeoutMS: 60000
           }
-        }),
-        { safe: true }
```

Comment:
The `safe` option of the `Db` constructor was deprecated a long time ago (in driver version 1: https://web.archive.org/web/20130408033926/http://mongodb.github.io:80/node-mongodb-native/api-generated/db.html) :older_woman:

Reply:
I know it was removed from the documentation in the Node driver _long ago_, but I'm concerned that this still isn't a no-op.  Unfortunately, I can't find the exact scenario where I looked into this, but I do remember digging into this in the past (with a similar revelation around the seeming deprecation of the `safe` option) long ago and being concerned by the fact that there seemed to be some support for it still in the connection string URI ([for example](https://github.com/mongodb/node-mongodb-native/search?utf8=%E2%9C%93&q=safe&type=)).

The `safe` option once related to the write-concern, but I'm not completely confident in removing this if it's possible it's still being passed through to Mongo itself in some way.  Can we confirm that it's not affecting the replica set configuration in any way (for example, does Mongo directly copy these to the connection string somewhere?) Do we need to more explicitly set a write-concern if we remove this? Should that be left as an implementation exercise to the developer?

Reply:
Ok, it appears that there's no way that `safe` is heading upstream to the core Mongo driver.  I've concluded this based on [this line](https://github.com/mongodb/node-mongodb-native/blob/b8484257e16ab129b5d1614f6c9fbf933846814d/lib/db.js#L162
) in the Mongo Node driver `Db` instantiation which uses a `filterOptions` with a so-called "`legalOptions`" whitelist which is [defined here](https://github.com/mongodb/node-mongodb-native/blob/b8484257e16ab129b5d1614f6c9fbf933846814d/lib/db.js#L82-L111) and has no sign of `safe`.

Reply:
That's good to know! I totally overlooked the `filterOptions` call. :eyes:

I'm not sure if I'm interpreting the [docs](https://docs.mongodb.com/manual/release-notes/drivers-write-concern/) and the driver's commit history correctly, but it looks like the `safe` option was replaced with `w = 1`, which is now the default write concern, so all operations should be "safe" automatically.